packages/react-components/react-color-picker-preview/library/src/components/ColorSlider/ColorSlider.types.ts

```diff
@@ -1,17 +1,71 @@
-import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import * as React from 'react';
+import type { ComponentProps, ComponentState, Slot, EventHandler, EventData } from '@fluentui/react-utilities';
+
+export type SliderOnChangeEventHandler = EventHandler<SliderOnChangeData>;
+
+export type SliderOnChangeData = EventData<'change', React.ChangeEvent<HTMLInputElement>> & {
+  value: number;
+};
 
 export type ColorSliderSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'div'>>;
+  rail: NonNullable<Slot<'div'>>;
+  thumb: NonNullable<Slot<'div'>>;
+  input: NonNullable<Slot<'input'>> & {
```

Comment:
Is this a component prop/attribute? Should this be part of the Slots definition?

Reply:
I asked Micah (as it was part of the Slider component). It's for internal use only, so I'll remove it